### PR TITLE
refact: example apps file structure and ts support

### PR DESCRIPTION
**Pull request by alduzy**
## Description

This PR intents to simplify example and test example apps file structure and fix typescript support.

## Changes

- removed package.json inside common dir and adjusted metro config to work with those dependencies
- relocated app.js file inside test examples common dir to match examples dir structure
- added submodules paths to the root tsconfig and extended it in the example dirs

## Test code and steps to reproduce

- Build and check wether the example and test example app works well on both paper and fabric architectures.
- Have a look at the imports in any file inside apps/ directory and enjoy less red marks + full typescript support for linked packages!

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Updated documentation: 
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/guides/GUIDE_FOR_LIBRARY_AUTHORS.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/native-stack/README.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/native-stack/types.tsx
- [ ] Ensured that CI passes

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,10 +2,25 @@
...
+        "link:../react-navigation/packages/elements/src"
+      ],
+      "@react-navigation/native": ["link:../react-navigation/packages/native/"],
+      "@react-navigation/native-stack": [
+        "react-navigation/packages/native-stack/src"
+      ],
+      "@react-navigation/routers": [
+        "link:../react-navigation/packages/routers/src"
+      ],
+      "@react-navigation/stack": ["link:../react-navigation/packages/stack/"]
```
Comment (kkafar) [inline tsconfig.json:23]: Awesome :tada:

**Review by kkafar — APPROVED** [review]
Hey, provided that it works (the apps do build and metro bundles) I think we are 🟢 

Thanks!

File: tsconfig.json [hunk 2]
```diff
@@ -2,10 +2,25 @@
...
+      "@react-navigation/bottom-tabs": [
+        "link:../react-navigation/packages/bottom-tabs/src"
+      ],
+      "@react-navigation/core": ["link:../react-navigation/packages/core/"],
+      "@react-navigation/elements": [
+        "link:../react-navigation/packages/elements/src"
+      ],
+      "@react-navigation/native": ["link:../react-navigation/packages/native/"],
+      "@react-navigation/native-stack": [
+        "react-navigation/packages/native-stack/src"
```
Comment (kkafar) [inline tsconfig.json:18]: Only thing that bothers me is why don't we use `link: ...` here as everywhere else?

File: FabricExample/package.json [hunk 3]
```diff
@@ -39,6 +40,7 @@
     "@react-native/metro-config": "0.74.83",
     "@react-native/typescript-config": "0.74.83",
     "@types/react": "^18.2.72",
+    "@types/react-native-restart": "^0.0.0",
```
Comment (WoLewicki) [inline FabricExample/package.json:43]: Weird version 🤔

File: tsconfig.json (hunk 2 above)
Comment (alduzy) [reply tsconfig.json]: seems not necessary at all, removed it: https://github.com/software-mansion/react-native-screens/pull/2174/commits/019d0fe79e5d1ceab2ab0bf854cc76c93d9314b6

File: FabricExample/package.json (hunk 3 above)
Comment (alduzy) [reply FabricExample/package.json]: You're right, I copied dependencies from the removed package.json to ensure nothing's missing but this types package looks strange. The [react-native-restart ](https://www.npmjs.com/package/react-native-restart) package offers should be ts ready by itself
[refact:removed unused types](https://github.com/software-mansion/react-native-screens/pull/2174/commits/bc6e06a6b444e219ff029d18ef12441391995686)

File: tsconfig.json (hunk 1 above)
Comment (alduzy) [reply tsconfig.json]: Simplified it by a lot in the recent commit, check it out: https://github.com/software-mansion/react-native-screens/pull/2174/commits/9a6662e0895bc98a6509ac821f045c45e96f1e12

**Review by kkafar — APPROVED** [review]
Fine for me

**Review by WoLewicki — APPROVED** [review]
LGTM, just test it with all caches cleared to be sure it works correctly :tada:

**Comment by kkafar** [comment]
I've triggered CI again as e2e Android tests & iOS build on Fabric have failed.

**Comment by alduzy** [comment]
The failing CI check `Test Android e2e / test` has been successfully tested by me and @kkafar on local. I'm going to re-run this check once again in the evening, but it seems to be come github CI fallacy.

**Review by maciekstosio — APPROVED** [review]

**Comment by alduzy** [comment]
It's worth noticing that it's not always the same test case that fails: It's either `e2e/examplesTests/events.e2e.ts`[here](https://github.com/software-mansion/react-native-screens/actions/runs/9446913169/job/26048415650?pr=2174) or `e2e/examplesTests/bottomTabs.e2e.ts` [here](https://github.com/software-mansion/react-native-screens/actions/runs/9418195335/job/26013810632) failing independently

**Review by kkafar — APPROVED** [review]
Yeah, I think facing the fact that the tests do pass w/o issues in local env, I think we should not be blocked by our CI setup / just CI runners and proceed here.

We need to handle these CI issues somehow separately.

File: Example/metro.config.js [hunk 4]
```diff
@@ -50,12 +50,6 @@ const config = {
     // to various errors. To mitigate this we define below custom request resolver, hijacking requests to conflicting modules and manually
     // resolving appropriate files. **Most likely** this can be achieved by proper usage of blockList but I found this method working ¯\_(ツ)_/¯
     resolveRequest: (context, moduleName, platform) => {
-      if (moduleName.startsWith('@react-navigation')) {
-        // For some reason, react-navigation packages don't want to resolve from
-        // the project's node_modules, so we need to use standard Metro resolver.
-        return context.resolveRequest(context, moduleName, platform);
-      }
-
```
Comment (tboba) [inline Example/metro.config.js:58]: Why this has been removed? Were there any errors regarding resolving this module? I'm just curious 🤔

Comment (alduzy) [reply Example/metro.config.js:58]: After deleting the package.json file from `apps/examples` this code had to be deleted as the app couldn't find the react navigation packages

**Pull request merged**